<launch>

    <arg default="false" name="multi" />
    <arg default="uav_" name="ns_prefix" />
    <arg default="sitl" name="mode" />
    <arg default="map" name="pose_frame_id" />
    <arg default="on" name="ual_server" />
    <arg default="30.0" name="ual_pub_rate" />
    <arg default="30.0" name="mavros_offboard_rate" />

    <arg default="true" name="ual_use" />
    <arg default="px4" name="autopilot" />
    <arg default="true" name="uav_manager_on_gs" />
    <arg default="false" name="smooth_path_generator" />

    <group if="$(eval mode == 'sitl')" ns="$(arg ns_prefix)3">
        <group>
            <group if="$(eval autopilot == 'px4')">
                <include file="$(find magna)/launch/PX4_spawn_robot_JA.launch">
                    <arg name="id" value="3" />
                    <arg name="material" value="Orange" />
                    <arg name="mode" value="$(arg mode)" />
                    <arg name="x" value="0" />
                    <arg name="y" value="0" />
                    <arg name="z" value="0" />
                    <arg name="yaw" value="0" />
                </include>
            </group>

            <group if="$(eval autopilot == 'dji')">
                <include file="$(find dji_sdk)/launch/sdk.launch" />
            </group>

            <group if="$(eval autopilot == 'bitcraze')">
                <include file="$(find crazyswarm)/launch/hover_swarm.launch" />
            </group>
        </group>
    

        <group if="$(arg ual_use)">
            <group if="$(eval autopilot == 'px4')">
                <node name="ual" output="screen" pkg="uav_abstraction_layer" type="server">
                    <param name="uav_id" value="3" />
                    <param name="pose_frame_id" value="$(arg pose_frame_id)" />
                    <param name="ual_server" value="$(arg ual_server)" />
                    <param name="ual_pub_rate" value="$(arg ual_pub_rate)" />
                    <param name="mavros_offboard_rate" value="$(arg mavros_offboard_rate)" />
                    <param name="home_pose_parent_frame" value="map" />
                    <param name="backend" value="mavros" />
                    <rosparam param="map_origin_geo">[37.558542, -5.931074, 0.0]</rosparam>
                </node>
            </group>
            <group if="$(eval autopilot == 'dji')">
                <node name="ual" output="screen" pkg="uav_abstraction_layer" type="server">
                    <param name="backend" value="dji" />
                    <param name="laser_altimeter" value="false" />
                    <param name="self_arming" value="true" />
                    <rosparam param="map_origin_geo">[37.4120687, -6.0021959, 71.5408066875]</rosparam>
                </node>
            </group>   
            <group if="$(eval autopilot == 'crazyflie')">
                <node name="ual" output="screen" pkg="uav_abstraction_layer" type="server">
                    <param name="backend" value="crazyflie" />
                </node>
            </group>
        </group>
    </group>

    <group if="$(arg uav_manager_on_gs)">
        <node args="-ID=3" name="agent_3" output="screen" pkg="magna" type="Agent_Manager.py" />
    </group>

</launch>